packages/next-auth/src/providers/shopify.js

```diff
@@ -0,0 +1,79 @@
+export default function Shopify(options) {
+  const {
+    shop = "",
+    scope = "",
+    clientId = "",
+    clientSecret = "",
+    apiVersion = "",
+  } = options
+
+  return {
+    id: "shopify",
+    name: "Shopify",
+    type: "oauth",
+    authorization: {
+      url: `https://${shop}.myshopify.com/admin/oauth/authorize`,
+      params: {
+        scope,
+      },
+    },
+    token: {
+      request: async (context) => {
+        if (!context.params.code)
+          throw new Error("No code returned by shopify server")
+        const payload = JSON.stringify({
+          client_id: clientId,
+          client_secret: clientSecret,
+          code: context.params.code,
+        })
+        const res = await fetch(
+          `https://${shop}.myshopify.com/admin/oauth/access_token`,
+          {
+            method: "POST",
+            headers: {
+              Accept: "application/json",
+              "Content-Type": "application/json",
+            },
+            body: payload,
+          }
+        )
+        if (res.ok) {
+          const result = await res.json()
+          return { tokens: result }
+        }
+
+        throw new Error(
+          "Something went wrong while trying to get the access_token"
+        )
+      },
+    },
```

Comment:
```suggestion
    client: {
      token_endpoint_auth_method: "client_secret_post",
    },
    token: `https://${shop}.myshopify.com/admin/oauth/access_token`,
```

This is a standard, so no need to reach for `request`.